specification/resources/resource-manager/Microsoft.Authorization/stable/2023-04-01/policySetDefinitionVersions.json

```diff
@@ -0,0 +1,915 @@
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/listPolicySetDefinitionVersions": {
+      "post": {
+        "tags": [
+          "PolicySetDefinitionVersions"
+        ],
+        "operationId": "PolicySetDefinitionVersions_ListAll",
+        "summary": "Lists all policy set definition versions within a subscription.",
+        "description": "This operation lists all the policy set definition versions for all policy set definitions within a subscription.",
+        "x-ms-examples": {
+          "List all policy definition versions at subscription": {
+            "$ref": "./examples/listAllPolicySetDefinitionVersions.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns an array of policy set definition versions.",
+            "schema": {
+              "$ref": "#/definitions/PolicySetDefinitionVersionListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Authorization/policySetDefinitions/{policySetDefinitionName}/versions/{version}": {
+      "put": {
+        "tags": [
+          "PolicySetDefinitionVersions"
+        ],
+        "operationId": "PolicySetDefinitionVersions_CreateOrUpdate",
+        "summary": "Creates or updates a policy set definition version.",
+        "description": "This operation creates or updates a policy set definition version in the given subscription with the given name and version.",
+        "x-ms-examples": {
+          "Create or update a policy set definition version": {
+            "$ref": "./examples/createOrUpdatePolicySetDefinitionVersion.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "policySetDefinitionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "pattern": "^[^<>*%&:\\?.+/]*[^<>*%&:\\?.+/ ]+$",
+            "description": "The name of the policy set definition."
+          },
+          {
+            "name": "version",
```

Comment:
>version [](http://example.com/codeflow?start=21&length=7)

Consider adding a pattern or at least add details to the description. Comment applies elsewhere in this PR as well (or consider factoring this parameter out to the "parameters" section below.